sources/main.cpp

```diff
@@ -267,7 +281,12 @@ std::pair<filesystem::path, hyde::json> load_hyde_config(
 
 /**************************************************************************************************/
 
-std::vector<std::string> integrate_hyde_config(int argc, const char** argv) {
+struct command_line_args {
+    std::vector<std::string> _hyde;
+    std::vector<std::string> _clang;
+};
+
+command_line_args integrate_hyde_config(int argc, const char** argv) {
```

Comment:
I split the command line arguments into two sets: those used for hyde, and those to be forwarded to the clang driver. This makes things a little bit cleaner below as the tool is setting up both itself and clang-tooling.